manager/manager.go

```diff
@@ -782,6 +782,10 @@ func (m *manager) createContainer(containerName string) error {
 		return err
 	}
 
+	if collectorConfigs != nil {
+		contSpec.HasCustomMetrics = true
+	}
```

Comment:
Oh, right! The spec gets re-written. Thanks for poiniting it out! I shall close the PR.